crates/pubsub/src/connect.rs

```diff
@@ -9,6 +9,16 @@ pub trait PubSubConnect: Sized + Send + Sync + 'static {
     /// Returns `true` if the transport connects to a local resource.
     fn is_local(&self) -> bool;
 
+    /// Select the buffer size for subscription channels. This value will
+    /// be used to determine the capacity of the broadcast channels used to send
+    /// notifications to subscribers. See [`broadcast::channel`] for more
+    /// information about channel behavior.
+    ///
+    /// [`broadcast::channel`]: tokio::sync::broadcast::channel
+    fn subscription_buffer_size(&self) -> usize {
+        16
+    }
```

Comment:
is only used by the `PubSubService`?

would it be better to move this to the PubSubFrontEnd? so it can be modified for future subscriptions?

for example:
you notice a lot of lagging responses and want to reconnect with larger buffer

Reply:
right now pubsub frontend and the service communicate only over the channel

we could do one of the following easily
- have pubsub frontend pass updates over the channel to the service, instructing it to change its conf value
- introduce shared memory between all frontends and the service

we can do the following messily but effectively:
- attach the conf value to each request, and have the backend store that conf value on a per-request basis

thoughts on which you like?

Reply:
per request would be best imo

Reply:
would you prefer that this is a single configuration value shared by all front ends and shared memory or a per front end configuration value?

Reply:
i want to clean up a bit by taking it off the inflight and making a `PubSubInstruction::Subscribe(Inflight, channel_size)`

Reply:
> i want to clean up a bit by taking it off the inflight and making a PubSubInstruction::Subscribe(Inflight, channel_size)

nvm putting it on the IX would require storing it int he backend separately. so i'll leave it on the inflight